component/storage/edv/restclient.go

```diff
@@ -230,6 +230,32 @@ func (c *restClient) deleteDocument(vaultID, docID string) error {
 	return fmt.Errorf(failResponseFromEDVServer, statusCode, respBytes)
 }
 
+func (c *restClient) addIndex(vaultID string, attributeNames []string) error {
+	addIndexOperation := indexOperation{
+		Operation:      "add",
+		AttributeNames: attributeNames,
+	}
+
+	jsonToSend, err := json.Marshal(addIndexOperation)
+	if err != nil {
+		return fmt.Errorf("failed to marshal index operation: %w", err)
+	}
+
+	endpoint := fmt.Sprintf("%s/%s/index", c.edvServerURL, url.PathEscape(vaultID))
+
+	statusCode, _, respBytes, err := c.sendHTTPRequest(http.MethodPost, endpoint, jsonToSend, c.headersFunc)
+	if err != nil {
+		return err
```

Comment:
It would be good to wrap the error with additional info, e.g.

`return fmt.Errorf("send HTTP request: %w", err)`

Sometimes when you look at an error in the log, you don't know which path the error took. :)

Reply:
Done!